### PR TITLE
(add) Documentation for new contributors to easily get started

### PR DESCRIPTION
**Pull request by valens200**
# Pull Requests Requirements

## Summary

Getting started Guide for onbarding GSoC'24 prospective contributors in openElis Global 2.

We understand how painful it might be for new onboarding contributors to get started to openElis Global 2 , As the project continues to evolve, it’s essential to ensure that our documentation remains up-to-date and accessible. We understand that navigating a new project can be challenging, especially for prospective contributors. Therefore, this documentation serves as a central hub for all information related to our project, making it easier for newcomers to get started.
## Screenshots

Screen Shot 2024-03-21 at 18 15 17

### CODE REVIEW

File: guide_docs.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+## Introduction
+![](https://)
+
+Getting started Guide for onbarding GSoC'24 prospective contributors in `openElis Global 2`.
+
+We understand how painful it might be for new onboarding contributors to get started to `openElis Global 2` , As the project continues to evolve, it’s essential to ensure that our documentation remains up-to-date and accessible. We understand that navigating a new project can be challenging, especially for prospective contributors. Therefore, this documentation serves as a central hub for all information related to our project, making it easier for newcomers to get started.
+
+## OpenELIS Global 2
+
+This laboratory information management tool rewrite onto Java Spring, and with all new technology and features. Please see our [website](http://www.openelis-global.org/) for more information.
```
Comment (HerbertYiga) [inline guide_docs.md:18]: i would prefer some thing like (This is a ....

Comment (valens200) [reply guide_docs.md:18]: Oooooh, let me look at this.

Comment (valens200) [reply guide_docs.md]: > i would prefer some thing like (This is a ....) when starting the sentence

@HerbertYiga the type is now resolved and I added other parts.
Thank you!

File: src/main/java/org/ozeki/sms/dao/OzekiMessageOutDAO.java [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package org.ozeki.sms.dao;
-
 import org.ozeki.sms.valueholder.OzekiMessageOut;
-import org.springframework.data.repository.CrudRepository;
+//import org.springframework.data.repository.CrudRepository;
```
Comment (sherrif10) [inline src/main/java/org/ozeki/sms/dao/OzekiMessageOutDAO.java:3]: Was this part of the ticket ? Seems to me like there couple of changes introduced not related to the ticket.

Comment (valens200) [reply src/main/java/org/ozeki/sms/dao/OzekiMessageOutDAO.java:3]: Hello @sherrif10, you. are right, that was not part of the ticket. What happened is this, I had changes in the previous branch and those changes are still coming  newly created branch related to GSoC documentation

Comment (valens200) [reply src/main/java/org/ozeki/sms/dao/OzekiMessageOutDAO.java:3]: And also @sherrif10 , I think that won't result into any problem

Comment (valens200) [reply src/main/java/org/ozeki/sms/dao/OzekiMessageOutDAO.java:3]: You are right @sherrif10, that was not part of the ticket.   What happened is this, I had changes in the previous branch and those changes are still coming newly created branch related to GSoC documentation.

But @sherrif10 I think that won't result in any problem as long as I don't make changes to that.

File: google-summer-of-code-2024.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+We prioritize maintaining high code quality through unit, integration, and end-to-end tests. OpenELIS utilizes various testing tools, including:
+
+* JUnit 5 (https://junit.org/junit5/docs/current/user-guide/)
+* Mockito (https://site.mockito.org/)
+* TestNG (https://testng.org/)
+* Cypress (https://www.cypress.io/)
+
+
+It is high recommended to rebuild the application and make sure your test cases are passing before opening any pull request.
```
Comment (Ankit-Matth) [inline google-summer-of-code-2024.md:242]: "It is **high** recommended....." should be "It is **highly** recommended........."

**Review by Ankit-Matth — CHANGES_REQUESTED** [review]
Hey @valens200, I found some spelling mistakes; please make sure to correct them.

**Comment by valens200** [comment]
> Hey @valens200, I found some spelling mistakes; please make sure to correct them.

Thank you brother @Ankit-Matth

**Pull request closed without merging**